### PR TITLE
Add Bastwifi by Electronic Cats

### PR DESCRIPTION
**Pull request by sabas1080**
Hi Circuit Python Team

We have added our new board Bast WiFi with ESP32S2 https://electroniccats.com/store/bast-wifi/

Thanks

### CODE REVIEW

File: ports/esp32s2/boards/electroniccats_bastwifi/mpconfigboard.mk [hunk 1]
```diff
@@ -0,0 +1,19 @@
+USB_VID = 0x239A
+USB_PID = 0x80A6
```
Comment (jepler) [inline ports/esp32s2/boards/electroniccats_bastwifi/mpconfigboard.mk:2]: This re-uses the PID assigned to the espressif_saola_1_wrover.  It should be a distinct PID, see #2966 and https://learn.adafruit.com/how-to-add-a-new-board-to-circuitpython/customizing-the-board-files#mpconfigboard-dot-mk-3033646-25.
> USB VID and PID are something you cannot borrow from another board. So don't copy and paste these values! If you don't have a VID/PID pair, open up an issue in CircuitPython repository and request one, we will assist in the process of assigning a PID for use with CircuitPython.

Comment (sabas1080) [reply ports/esp32s2/boards/electroniccats_bastwifi/mpconfigboard.mk]: Oh sorry, fixed

**Comment by jepler** [comment]
This should be good to merge after https://github.com/pidcodes/pidcodes.github.com/pull/542 makes the PID allocation official.

File: ports/esp32s2/boards/electroniccats_bastwifi/mpconfigboard.mk (hunk 1 above)
Comment (jepler) [reply ports/esp32s2/boards/electroniccats_bastwifi/mpconfigboard.mk]: Thank you!

**Comment by jepler** [comment]
@tannewt pidcodes seems very slow to act, no github activity since April. I had suggested waiting for them to act, but should we just merge this anyway?

**Comment by jepler** [comment]
@tannewt giving you some fresh mention-spam

**Comment by tannewt** [comment]
Ok, I've pinged the pid.codes person to see if they need help doing reviews for the repo. (I'd be happy to help review.)

**Comment by dhalbert** [comment]
Letting this hang until pidcodes approves seems like a mistake, given that pidcodes has gone dormant. Since it's in a PR to pidcodes, and presumably no one else has asked for the same PID (maybe double-check) in another PR, I think we could go ahead and approve. Sound ok to you, @tannewt?

**Comment by dhalbert** [comment]
Merged this for now; if necessary we can change the VID/PID later.

**Comment by dhalbert** [comment]
This broke in the build: https://github.com/adafruit/circuitpython/runs/1042094212?check_suite_focus=true. Maybe it's due to ESP-IDF changes?? I will revert this out for now.

@sabas1080 could you take a look?

**Comment by tannewt** [comment]
Note that the build breakage wasn't caused by this PR. It is flakiness in the ESP build.

Fine with me to merge too.

**Comment by sabas1080** [comment]
Thanks @tannewt @dhalbert

**Pull request merged**